src/avro_reader.cpp

```diff
@@ -345,28 +322,51 @@ static void TransformValue(avro_value *avro_val, const AvroType &avro_type, Vect
 		break;
 	}
 
+	case LogicalTypeId::MAP:
 	case LogicalTypeId::LIST: {
 		size_t list_len;
-
 		if (avro_value_get_size(avro_val, &list_len)) {
 			throw InvalidInputException(avro_strerror());
 		}
-		auto &child_vector = ListVector::GetEntry(target);
+
 		auto child_offset = ListVector::GetListSize(target);
 		ListVector::Reserve(target, child_offset + list_len);
 
-		for (idx_t child_idx = 0; child_idx < list_len; child_idx++) {
-			avro_value_t child_value;
-			if (avro_value_get_by_index(avro_val, child_idx, &child_value, nullptr)) {
-				throw InvalidInputException(avro_strerror());
+		if (avro_type.avro_type == AVRO_ARRAY) {
+			auto &child_vector = ListVector::GetEntry(target);
+
+			for (idx_t child_idx = 0; child_idx < list_len; child_idx++) {
+				avro_value_t child_value;
+				if (avro_value_get_by_index(avro_val, child_idx, &child_value, nullptr)) {
+					throw InvalidInputException(avro_strerror());
+				}
+				TransformValue(&child_value, avro_type.children[0].second, child_vector, child_offset + child_idx);
+			}
+		} else {
+			auto &key_vector = MapVector::GetKeys(target);
```

Comment:
This looks confusing, but avro can either use the `map` type, which always has string keys, or (this might be iceberg specific, but eh) it can use `array` with the `"logicalType": "map"` metadata, to indicate that it has a "record" with fields "key", "value" as `items` of the array (in which case it gets handled above)